spec/rubocop/cop/gemspec/required_ruby_version_spec.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+describe RuboCop::Cop::Gemspec::RequiredRubyVersion, :config do
+  subject(:cop) { described_class.new(config) }
+
+  before do
+    allow(File).to receive(:basename).with('(string)').and_return('foo.gemspec')
```

Comment:
https://github.com/bbatsov/rubocop/blob/b7a291ab441e415e5657bf92710e4644803e604a/lib/rubocop/rspec/expect_offense.rb#L47
Can you try passing file name as `expect_offense`'s 2nd argument instead of stub? It probably works well, and it does not magical.